components/tooltip/Tooltip.js

```diff
@@ -28,203 +28,201 @@ const SIZES = {
-  return ComposedComponent => {
-    class TooltippedComponent extends Component {
-      static propTypes = {
-        children: PropTypes.node,
-        className: PropTypes.string,
-        onClick: PropTypes.func,
-        onMouseEnter: PropTypes.func,
-        onMouseLeave: PropTypes.func,
-        tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-        tooltipColor: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'inverse']),
-        tooltipHideOnClick: PropTypes.bool,
-        tooltipIcon: PropTypes.element,
-        tooltipPosition: PropTypes.oneOf(Object.keys(POSITIONS).map(key => POSITIONS[key])),
-        tooltipShowOnClick: PropTypes.bool,
-        tooltipSize: PropTypes.oneOf(Object.keys(SIZES)),
-        documentObject: PropTypes.object.isRequired,
+const Tooltip = ComposedComponent => {
+  class TooltippedComponent extends Component {
+    static propTypes = {
+      children: PropTypes.node,
+      className: PropTypes.string,
+      onClick: PropTypes.func,
+      onMouseEnter: PropTypes.func,
+      onMouseLeave: PropTypes.func,
+      tooltip: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+      tooltipColor: PropTypes.oneOf(['white', 'neutral', 'mint', 'violet', 'ruby', 'gold', 'aqua', 'inverse']),
+      tooltipHideOnClick: PropTypes.bool,
+      tooltipIcon: PropTypes.element,
+      tooltipPosition: PropTypes.oneOf(Object.keys(POSITIONS).map(key => POSITIONS[key])),
+      tooltipShowOnClick: PropTypes.bool,
+      tooltipSize: PropTypes.oneOf(Object.keys(SIZES)),
+      documentObject: PropTypes.object.isRequired,
+    };
+
+    static defaultProps = {
+      className: '',
+      tooltipColor: 'white',
+      tooltipHideOnClick: true,
+      tooltipIcon: null,
+      tooltipPosition: POSITIONS.TOP,
+      tooltipShowOnClick: false,
+      tooltipSize: 'medium',
+    };
+
+    constructor(props) {
+      super(...arguments);
+
+      this.state = {
+        active: false,
+        position: this.props.tooltipPosition,
       };
 
-      static defaultProps = {
-        className: '',
-        tooltipColor: 'white',
-        tooltipHideOnClick: true,
-        tooltipIcon: null,
-        tooltipPosition: POSITIONS.TOP,
-        tooltipShowOnClick: false,
-        tooltipSize: 'medium',
-      };
+      this.tooltipRoot = props.documentObject.createElement('div');
+      this.tooltipRoot.id = 'tooltip-root';
+    }
 
-      constructor(props) {
-        super(...arguments);
+    getPosition(element) {
+      const { tooltipPosition } = this.props;
 
-        this.state = {
-          active: false,
-          position: this.props.tooltipPosition,
-        };
-
-        this.tooltipRoot = props.documentObject.createElement('div');
-        this.tooltipRoot.id = 'tooltip-root';
-      }
+      if (tooltipPosition === POSITIONS.HORIZONTAL) {
+        const origin = element.getBoundingClientRect();
+        const { width: windowWidth } = getViewport();
```

Comment:
PP: What do you think of moving `const origin` and `const { ... } = getViewPort()` outside of the if block? It looks like the result of those is used in both the `if` and `else` blocks.